### PR TITLE
validate private keys

### PR DESCRIPTION
**Pull request by rrrooommmaaa**
This PR validates all private key packets on `decrypt` or on `parse` (if they are unencrypted)

close #0000 // if this PR closes an issue

----------------------------------

**Tests** _(delete all except exactly one)_:
- Tests added or updated

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [ ] addresses the issue it closes (if any)
- [ ] code is readable and understandable
- [ ] is accompanied with tests, or tests are not needed
- [ ] is free of vulnerabilities
- [ ] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

File: test/source/tests/unit-node.ts [hunk 1]
```diff
@@ -1634,6 +1634,142 @@ jA==
       t.pass();
     });
 
+    ava.default('[unit][KeyUtil.decrypt] validates the private key', async t => {
```
Comment (tomholub) [inline test/source/tests/unit-node.ts:1637]: This looks good. Does this also fix the issue as experienced when private key is loaded:
 - from inbox backup
 - from EKM

**Review by tomholub — COMMENTED** [review]
:+1:

File: test/source/tests/unit-node.ts (hunk 1 above)
Comment (tomholub) [reply test/source/tests/unit-node.ts:1637]: In other words, is this a full / verified fix for the problem?

**Comment by tomholub** [comment]
@martgil I think you had issue reproducing the problem. You can ask here. Did you try to put the broken key into inbox and then load it? If it succeeded, then with the above PR it should fail - please triage.

File: test/source/tests/unit-node.ts (hunk 1 above)
Comment (rrrooommmaaa) [reply test/source/tests/unit-node.ts:1637]: Let me add validation for `KeyUtil.readBinary()` too...

**Comment by martgil** [comment]
> Did you try to put the broken key into inbox and then load it?

Yes, that's exactly what I did.

>  If it succeeded, then with the above PR it should fail - please triage.

Yes @tomholub, the private key is blocked and no longer accepted when a user tried to import/add it.

Screen Shot 2022-01-27 at 2 02 52 PM
Screen Shot 2022-01-27 at 2 03 37 PM

File: test/source/tests/unit-node.ts (hunk 1 above)
Comment (rrrooommmaaa) [reply test/source/tests/unit-node.ts:1637]: Done

**Comment by tomholub** [comment]
@martgil yes, please file the issues there. For Android, ping @IvanPizhenko so that he can maybe create a test case for Paul to look at.

**Review by tomholub — APPROVED** [review]

**Pull request merged**